### DISCUSSION TITLE
What all can this project track?

### DISCUSSION THREAD

**Discussion by IgnisDa** [IgnisDa/ryot#4] [Ideas]
The following is a list of all things that I want this application to be able to keep track of.

## Media

- [X] Books (via [Openlibrary](openlibrary.org/))
- [X] Movies (via [TMDB](https://www.themoviedb.org/))
- [X] Shows (via [TMDB](https://www.themoviedb.org/))
- [X] Video Games (via [IGDB](https://igdb.com))
- [x] Audio Books (via [Audible](https://audible.readthedocs.io/en/latest/misc/external_api.html))
- [x] Podcasts (via [Listen Notes](https://www.listennotes.com/))
- [x] Manga (via [Anilist](https://anilist.co/graphiql))
- [x] Anime (via [Anilist](https://anilist.co/graphiql))
- [x] Visual Novels (via [VnDB](https://vndb.org/))
- [ ] Music (via [ListenBrainz](https://listenbrainz.org/))

## Fitness

- [x] Gym exercises (via [Wrkout](https://github.com/wrkout/exercises.json), [Yuhonas](https://github.com/yuhonas/free-exercise-db))

## General

- [ ] Websites: time spent on a website

(Items marked as done are already supported).

If you have any suggestions, feel free to comment them.

**Comment by tomtom215** [comment]
I bet a lot of people would jump on this project if you support: 

- Plex & Tautulli (and Jellyfin)
- Sonarr & Radarr

**Reply by claythegamer** [reply]
This
If you could import from trakt it would be a huge plus!

**Reply by IgnisDa** [reply]
I have never used these services and hence did not find the need to support them. But seems like everyone want to see these integrations so I will look into it.

**Reply by C-Fu** [reply]
A lot of people who uses plex/jellyfin/emby and searches online about it will at some point discover the *arrs and tautulli/jellystat.

Also a support of subsonic-compatible server/apps would be great, since there are a billion of variations of these out there for pretty much everything including the kitchen sink (I use Navidrome btw).

**Comment by tomtom215** [comment]
For fitness, perhaps take a look at this project for reference which successfully integrates with Peloton and Garmin API's, that would be two large ecosystems for you to include: https://github.com/philosowaffle/peloton-to-garmin

**Reply by IgnisDa** [reply]
Thank you very much for your help!

**Comment by spiral6** [comment]
Anime and Manga (i.e. MAL, Anilist, etc).

**Reply by IgnisDa** [reply]
COuld you not found what you were looking for under `Books` and `Shows`?

**Reply by spiral6** [reply]
Neither are good enough scrapers/trackers that have every single anime or manga, particularly more obscure ones. Also doesn't apply to manhwa or webtoons.

**Reply by IgnisDa** [reply]
Alright. Makes sense to add these too.

**Reply by IgnisDa** [reply]
@spiral6 I have been looking into Anilist and MAL. Both of these do not provide information about an anime's individual episodes (or a manga's chapters). Is this acceptable to anime enthusiasts?

**Reply by spiral6** [reply]
I cannot speak to most enthusiasts, but for myself personally, I find that individual issues or episodes in anime and manga are less important than say, TV shows or individual songs on an album. Being unable to rate individual episodes is not a blocking factor for myself.

**Reply by IgnisDa** [reply]
I'm not talking about rating then. I'm talking about marking individual chapters/episode as seen.

**Reply by spiral6** [reply]
In Anilist, users cannot finitely/individually pick episodes and mark them as watched (as you'd see on IMDB or Trakt). 
Instead, they fill out a number indicating how many episodes they've watched, based on the total episodes they've watched.
![image](https://github.com/IgnisDa/ryot/assets/2887489/cf510fff-d907-4e34-82c7-899bf8b2ed05)
(The GraphQL also provides access to this, of course).

It's the same in MyAnimeList as well. I don't believe most people worry about the tracking of individual episodes.

**Reply by IgnisDa** [reply]
Alright makes sense. Thank you very much!

**Reply by IgnisDa** [reply]
@spiral6 `v1.2.0` supports both anime and manga!

**Reply by spiral6** [reply]
Thank you very much for the hard work! :)

**Comment by coolmule0** [comment]
### Media

  - [ ]  Music (via [Listenbrainz](https://listenbrainz.org/))

**Comment by modymp** [comment]
-- related to the Reddit comment > https://www.reddit.com/r/selfhosted/comments/13rehkg/comment/jlls3le/ - 

All the above listed seems to leverage API from other services, which seems to indicate an automated tracking approach. 

I have a few things that I would want to track that will be manual such as the below, 

1. classes that my kid/myself goes to (swimming, football, etc)   
2. daily personal habits to keep track of  
3. (shall update later)..

I have searched but couldnt find something decent for "my requirements" (ie., in terms decent UI/UX/features) in the selfhosted space. For example, https://habitica.com/ is cool but I dont intend to gamify ++ a few others too. (or, my searches were bad and I did not stumble on something yet) 

> a few that are hosted/serviced, that I really like    
https://present-app.com/ - a iOS app 
https://pixe.la/ - a nice one with API support for integration, graphs & heatmaps too   
https://www.dailyhabits.xyz/ - another simple one 
 
thanks!

**Reply by IgnisDa** [reply]
Could you take a look at #73. It might be what you need.

**Comment by drinksomewhiskey** [comment]
Is it possible to include price tracking via pricecharting.com for videogames? It would be nice to select condition as well (e.g., new, compete in box, loose).

**Reply by IgnisDa** [reply]
I have plans to add implemention of correlation between data from 2 different sources. Right now game data come from IGDB. Now let's say you want to add correlation of a game from IGDB to a item on pricecharting, you'll have to do that manually after which it downloads all the data and shows it on the UI.

Does this make sense?

**Reply by drinksomewhiskey** [reply]
Meaning, you would have to manually add a price charting ID after adding media to the library? Then, it will pull values afterwards?

**Reply by IgnisDa** [reply]
Yep that's correct. Since there is no way for Ryot to do so accurately itself.

**Comment by GinjNTonic** [comment]
Would it be possible to import existing lists from Letterboxd (movie tracker) accounts?  The data export from there is a csv file.

**Reply by IgnisDa** [reply]
We have #64 that will define a common interface for all imports. Then I plan to create scripts (or maybe docker images) to actually perform these imports. Having the import separately will ensure that the actual server remains as lean as possible.

**Comment by edersong** [comment]
For fitness, I would suggest a integration or a way to import data from OpenScale https://github.com/oliexdev/openScale

**Reply by IgnisDa** [reply]
This looks good. What exactly would an integration between the two entail?

**Comment by Batman313v** [comment]
I would love to see some type of food/ snack/ grocery tracker with a "ratings" system so I could easily look of something up for when I don't remember if my girlfriend and/ or I like that flavor/ meal/ option. Would be nice

# Suggestion
## Food / Snack / Meal
These could either be combined or placed under something like 
- Food
  - Snack
  - Meal
  
### Options
Some quick ideas of what it could include
- Ratings
  - Add ratings with Stars and details to easily see what was thought after the last time it was eaten
- Info
  - Add information for a "food item" like nutrition info, Calories (Maybe be implemented with fitness?)
  - Recipe
    - Steps in a check list
    - Ingredients
  - Upload Photos?

### Story
I'm not big into food so there are most definitly things that could be added to this that I havn't thought of but I thought of the idea while standing in the store and after realizing they were out of my favorite flavor of an item I couldn't remeber which of the other options I didn't like as to avoid accidentally getting that one. So when I got home I jumped on reddit in search of a solution and came across ryot. While not a solution for the food problem I loved the Idea of an all in one tracker so here I am adding a suggestion to the list 😁

**Reply by IgnisDa** [reply]
This sounds like a good idea? Is this something that #73 might be useful for?

**Reply by Batman313v** [reply]
I definitely think that this is a simple enough use case that it could be covered by that, I hadn't seen that yet but it's a really cool idea to have dynamic declarative config system for custom trackers. I just discovered this project today but am already super excited to see where it goes. Hopefully once I get more confident in Rust I can help contribute 👍

**Reply by IgnisDa** [reply]
Yes it has not been implemented yet (and won't be for a long time since I will be working on fitness tracking), but I am collecting feedback on it. I would love it if you leave yours in that issue.

Thanks for your interest in the project.

**Reply by lapius** [reply]
Try https://grocy.info

**Comment by C-Fu** [comment]
For Comic/Manga, Android users will definitely have Tachiyomi app on their phones. Not sure if you can somehow tap into the read status from there.

Comick website/service/app is also very very big and very very good (also a source in Tachiyomi).

For tracking time spent on websites, I'd suggest a chrome/firefox addon maybe?

Keep it up, a lot of people will need this!

**Reply by IgnisDa** [reply]
> Comick website/service/app is also very very big and very very good (also a source in Tachiyomi).

Do the existing media providers not work for you?

> For tracking time spent on websites, I'd suggest a chrome/firefox addon maybe?

Yes wakatime has a plugin for this. I plan to implement Ryot's add-ons similarly.

**Comment by el-tiuri** [comment]
Adding Strava integration would be huge for tracking exercises, that might be worth looking into.

**Reply by IgnisDa** [reply]
That's a good idea. Please make an issue for this.

**Comment by Dylancyclone** [comment]
Is it possible to use OpenLibrary as a source for Audiobooks in addition to Audible?
I've got a few audiobooks that have been removed from Audible that I would still like to track that I was able to find on OpenLibrary
Examples:
[The Dark Worlds of H.P. Lovecraft, Vol 1](https://www.goodreads.com/book/show/564315.The_Dark_Worlds_of_H_P_Lovecraft_Vol_1)
- [OpenLibrary](https://openlibrary.org/works/OL152251W/The_Dark_Worlds_Of_H._P._Lovecraft_Volume_1)
- [Audible](https://www.audible.com/pd/0809562677)

[I Have No Mouth and I Must Scream: The Voice from the Edge Vol. 1](https://www.goodreads.com/book/show/3839976-i-have-no-mouth-and-i-must-scream?ref=nav_sb_ss_1_65)
- [OpenLibrary](https://openlibrary.org/works/OL1737278W/I_Have_No_Mouth_and_I_Must_Scream_The_Voice_from_the_Edge_Vol._I_%28Fantastic_Audio_Series_the_Voice_f)
- [Audible](https://www.audible.com/pd/0787122661)

(I have the audible links because the audiobook's AISN number is saved in Audiobookshelf)

**Reply by IgnisDa** [reply]
Just log these as a book. Using Openlibrary as audiobooks provider makes no sense.

**Reply by Dylancyclone** [reply]
Sounds good, just wanted to check since I noticed Openlibrary differentiated between physical/audio editions

Thank you for your work on this project! It's been great so far

**Comment by dlucey1971** [comment]
For fitness, a way to upload/sync Heart Rate Monitor (HRM) information via .csv file to track your workout performance.
